sound/soc/sof/loader.c

```diff
@@ -75,8 +75,7 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 
 	while (ext_hdr->hdr.cmd == SOF_IPC_FW_READY) {
 		/* read in ext structure */
-		offset += sizeof(*ext_hdr);
-		snd_sof_dsp_block_read(sdev, bar, offset,
+		snd_sof_dsp_block_read(sdev, bar, offset + sizeof(*ext_hdr),
```

Comment:
this should probably be a separate PR, so that we can properly test it. It really has nothing to do with the compiler.

Reply:
Without this fix, my changes won't work.

Reply:
ok, will upstream this first then